t/plugin/jwt-auth-vault.t

```diff
@@ -0,0 +1,368 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+no_shuffle();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    my $http_config = $block->http_config // <<_EOC_;
+
+    server {
+        listen 8777;
+
+        location /secure-endpoint {
+            content_by_lua_block {
+                ngx.say("successfully invoked secure endpoint")
+            }
+        }
+    }
+_EOC_
+
+    $block->set_value("http_config", $http_config);
+
+    if (!$block->request) {
+        $block->set_value("request", "GET /t");
+    }
+    if (!$block->no_error_log && !$block->error_log) {
+        $block->set_value("no_error_log", "[error]\n[alert]");
+    }
+});
+
+run_tests;
+
+__DATA__
+
+=== TEST 1: schema check
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.jwt-auth")
+            local core = require("apisix.core")
+            for _, conf in ipairs({
+                {
+                    -- public and private key are not provided for RS256, returns error
+                    key = "key-1",
+                    algorithm = "RS256"
+                },
+                {
+                    -- public and private key are not provided but vault config is enabled.
+                    key = "key-1",
+                    algorithm = "RS256",
+                    vault = {}
+                }
+            }) do
+                local ok, err = plugin.check_schema(conf, core.schema.TYPE_CONSUMER)
+                if not ok then
+                    ngx.say(err)
+                else
+                    ngx.say("ok")
+                end
+            end
+        }
+    }
+--- response_body
+failed to validate dependent schema for "algorithm": value should match only one schema, but matches none
+ok
+
+
+
+=== TEST 2: create a consumer with plugin and username
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/consumers',
+                ngx.HTTP_PUT,
+                [[{
+                    "username": "jack",
+                    "plugins": {
+                        "jwt-auth": {
+                            "key": "key-hs256",
+                            "algorithm": "HS256",
+                            "vault":{}
+                        }
+                    }
+                }]],
+                [[{
```

Comment:
Could we remove the expected response? We don't need to check this output anymore. More other tests already do it.

Reply:
Sure. Done